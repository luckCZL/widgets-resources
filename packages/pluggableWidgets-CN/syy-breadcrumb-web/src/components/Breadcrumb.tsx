import { ReactElement, createElement } from "react";
import { Breadcrumb } from "antd";
import { ContentInterface } from "../BreadcrumbWeb";

export interface BreadcrumbProps {
    className: string;
    separator: string;
    style?: React.CSSProperties | undefined;
    data: ContentInterface[];
    onClick: (item: ContentInterface) => void;
    handleIcon: (icon: string) => void;
}

export function BreadcrumbCom(props: BreadcrumbProps): ReactElement {
    const { className, separator, style, data, onClick, handleIcon } = props;

    return (
        <Breadcrumb style={style} className={className} separator={separator}>
            {data.map(item => {
                return (
                    <Breadcrumb.Item key={item.id} onClick={() => onClick(item)}>
                        {item.icon && handleIcon(item.icon)}
                        <a href="javascript:;">{item.title}</a>
                    </Breadcrumb.Item>
                );
            })}
        </Breadcrumb>
    );
}
