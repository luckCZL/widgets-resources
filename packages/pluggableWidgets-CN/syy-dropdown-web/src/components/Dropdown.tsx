import { ReactElement, createElement, ReactNode } from "react";
import Dropdown from "antd/lib/dropdown";
import { TriggerEnum, PlacementEnum } from "../../typings/DropdownWebProps";

export interface DropdownProps {
    trigger: TriggerEnum;
    placement: PlacementEnum;
    arrow: boolean;
    overlay: () => JSX.Element;
    children: ReactNode;
}

export default function DropdownCom(props: DropdownProps): ReactElement {
    const { trigger, placement, arrow, children, overlay } = props;

    return (
        <Dropdown overlay={overlay} trigger={[trigger]} placement={placement} arrow={arrow}>
            {children}
        </Dropdown>
    );
}
