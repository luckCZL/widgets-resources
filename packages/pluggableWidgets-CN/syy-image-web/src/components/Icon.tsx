import { createElement, ReactElement } from "react";
import classNames from "classnames";
import { WebIcon } from "mendix";

interface IconProps {
    animate?: boolean;
    image?: boolean;
    full?: boolean;
    value: WebIcon;
}

export function Icon({ animate, image, full, value }: IconProps): ReactElement {
    if (value && value.type === "glyph") {
        return (
            <span
                className={classNames(
                    "rating-icon",
                    { "rating-icon-image": image, "rating-icon-full": full, animate },
                    "glyphicon",
                    value.iconClass
                )}
                aria-hidden="true"
            />
        );
    }
    if (value && value.type === "image") {
        return (
            <img
                className={classNames("rating-image", {
                    "rating-image-image": image,
                    "rating-image-full": full,
                    animate
                })}
                src={value.iconUrl}
                alt=""
            />
        );
    }
    return <div />;
}
