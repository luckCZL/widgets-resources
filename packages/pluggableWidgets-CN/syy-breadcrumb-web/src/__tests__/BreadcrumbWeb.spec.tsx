import { createElement } from "react";
// import { render } from "@testing-library/react-native";
import { shallow } from "enzyme";
// import { mount, ReactWrapper, shallow, ShallowWrapper } from "enzyme";
import { BreadcrumbCom, BreadcrumbProps } from "../components/Breadcrumb";
// import { BreadcrumbWebContainerProps } from "../../typings/BreadcrumbWebProps";

// const defaultProps: BreadcrumbWebContainerProps = {
//     name: "BreadcrumbWebTest",
//     class: "",
//     staticData: [],
//     isJump: false,
//     separator: "-"
// };

describe("BreadcrumbWeb", () => {
    let defaultProps: BreadcrumbProps;
    beforeEach(() => {
        defaultProps = {
            className: "BreadcrumbWebTest",
            data: [
                {
                    id: "1",
                    title: "一级菜单"
                },
                {
                    id: "2",
                    title: "二级菜单"
                }
            ],
            separator: "-",
            handClick: jest.fn(),
            handleIcon: jest.fn()
        };
    });

    it("renders as a breadcrumb", () => {
        const breadcrumb = shallow(<BreadcrumbCom {...defaultProps} />);
        expect(breadcrumb).toMatchSnapshot();
    });

    it("renders a default separator", () => {
        const breadcrumb = shallow(<BreadcrumbCom {...defaultProps} separator=":" />);
        expect(breadcrumb).toMatchSnapshot();
    });

    it("renders a default data", () => {
        const breadcrumb = shallow(<BreadcrumbCom {...defaultProps} data={[]} />);
        expect(breadcrumb).toMatchSnapshot();
    });

    // it("triggers handClick function with a click event", () => {
    //     defaultProps.handClick = jest.fn();
    //     // const onClick = jest.fn();
    //     const breadcrumb = mount(<BreadcrumbCom {...defaultProps} handClick={defaultProps.handClick} />);

    //     breadcrumb.find("BreadcrumbItem").simulate("click");

    //     expect(defaultProps.handClick).toHaveBeenCalledTimes(1);
    // });
});
