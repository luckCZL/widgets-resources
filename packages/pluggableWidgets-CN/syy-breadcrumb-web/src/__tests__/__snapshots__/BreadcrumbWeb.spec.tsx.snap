// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbWeb renders a default data 1`] = `
<Breadcrumb
  className="BreadcrumbWebTest"
  separator="-"
/>
`;

exports[`BreadcrumbWeb renders a default separator 1`] = `
<Breadcrumb
  className="BreadcrumbWebTest"
  separator=":"
>
  <BreadcrumbItem
    key="1"
    onClick={[Function]}
  >
    <a
      href="javascript:;"
    >
      一级菜单
    </a>
  </BreadcrumbItem>
  <BreadcrumbItem
    key="2"
    onClick={[Function]}
  >
    <a
      href="javascript:;"
    >
      二级菜单
    </a>
  </BreadcrumbItem>
</Breadcrumb>
`;

exports[`BreadcrumbWeb renders as a breadcrumb 1`] = `
<Breadcrumb
  className="BreadcrumbWebTest"
  separator="-"
>
  <BreadcrumbItem
    key="1"
    onClick={[Function]}
  >
    <a
      href="javascript:;"
    >
      一级菜单
    </a>
  </BreadcrumbItem>
  <BreadcrumbItem
    key="2"
    onClick={[Function]}
  >
    <a
      href="javascript:;"
    >
      二级菜单
    </a>
  </BreadcrumbItem>
</Breadcrumb>
`;
